lib/stream.js

```diff
@@ -45,37 +45,50 @@ Stream.Stream = Stream;
 try {
   const types = require('util').types;
   if (types && typeof types.isUint8Array === 'function') {
-    Stream._isUint8Array = types.isUint8Array;
+    // Stream._isUint8Array = types.isUint8Array;
+    Stream._isArrayBufferView = types.isArrayBufferView;
   } else {
     // This throws for Node < 4.2.0 because there's no util binding and
     // returns undefined for Node < 7.4.0.
-    Stream._isUint8Array = process.binding('util').isUint8Array;
+    // Stream._isUint8Array = process.binding('util').isUint8Array;
+    Stream._isArrayBufferView = process.binding('util').isArrayBufferView;
   }
 } catch (e) {
 }
 
-if (!Stream._isUint8Array) {
-  Stream._isUint8Array = function _isUint8Array(obj) {
-    return Object.prototype.toString.call(obj) === '[object Uint8Array]';
+if (!Stream._isArrayBufferView) {
+  Stream._isArrayBufferView = function _isArrayBufferView(obj) {
+    return [
+      '[object Int8Array]',
+      '[object Uint8Array]',
+      '[object Uint8ClampedArray]',
+      '[object Int16Array]',
+      '[object Uint16Array]',
+      '[object Int32Array]',
+      '[object Uint32Array]',
+      '[object Float32Array]',
+      '[object Float64Array]',
+      '[object DataView]'
+    ].includes(Object.prototype.toString.call(obj));
```

Comment:
This is going to be pretty slow. Surely there is a better way...

Reply:
Will `.indexOf()` be faster? Open to other suggestions as well.

Reply:
Can't we just use something like `ArrayBuffer.isView()`? It seems to be available since at least node v4. The reason the string check was used before was because we were explicitly checking for only `Uint8Array`.

Reply:
On `util.types` are a lot of checks that can be used as well.

Reply:
@BridgeAR the relevant function from `util.types` is already being used earlier in this file if it exists. This implementation here is used if either it or the function from the `util` binding are not available.

Reply:
@mscdex so `ArrayBuffer.isView()` could still be available, even though `util.types` and `util` binding are not available?
In other PRs dealing with typed arrays, i have been using:
```
const { isArrayBufferView } = require('internal/util/types');
```
Same can be used here too?
Wasn't sure whether compatibility with `0.x` versions was specifically required here, and how to add that support.

Reply:
So i'm guessing current PR stays put?

Or is there any merit in adding a patch of this sort:
```
diff --git a/lib/stream.js b/lib/stream.js
index c6861b3464..58aab7c16d 100644
--- a/lib/stream.js
+++ b/lib/stream.js
@@ -42,6 +42,15 @@ Stream.finished = eos;
 Stream.Stream = Stream;

 // Internal utilities
+try {
+  const { isArrayBufferView } = require('internal/util/types');
+  if (typeof isArrayBufferView === 'function') {
+    Stream._isArrayBufferView = isArrayBufferView;
+  }
+} catch (e) {
+}
+
+if (!Stream._isArrayBufferView) {
   try {
     const types = require('util').types;
     if (types && typeof types.isArrayBufferView === 'function') {
@@ -53,6 +62,7 @@ try {
     }
   } catch (e) {
   }
+}

 if (!Stream._isArrayBufferView) {
   Stream._isArrayBufferView = function _isArrayBufferView(obj) {
```

Reply:
@SirR4T That's more or less the exact same thing that's being done below that with `require('util').types`. I still think using `ArrayBuffer.isView()` as the fallback is probably best. I suppose the function could be cached at startup like `internal/util/types` does.

Reply:
Updated to use `ArrayBuffer.isView`. Although do verify, if simply having it in a constant will cache it? Or should the `const isTypedArray` be moved to module scope?

Also, shouldn't we be checking for existence of `ArrayBuffer.isView` first, and then be falling back to other methods (`util`.types)?